tree/ntuple/v7/src/RNTupleProcessor.cxx

```diff
@@ -415,25 +459,24 @@ ROOT::Experimental::RNTupleJoinProcessor::RNTupleJoinProcessor(
          auto valuePtr = fModel->GetDefaultEntry().GetPtr<void>(fieldName);
          fEntry->BindValue(fieldName, valuePtr);
       }
+   }
+
+   fPrimaryProcessor->SetEntryPointers(*fEntry);
+   for (auto &auxProcessor : fAuxiliaryProcessors) {
+      auxProcessor->SetEntryPointers(*fEntry, auxProcessor->GetProcessorName());
+   }
+
+   if (!joinFields.empty()) {
+      fJoinFieldValues.reserve(joinFields.size());
+      for (const auto &joinField : joinFields) {
+         auto token = fEntry->GetToken(joinField);
+         auto ptr = fEntry->GetPtr<void>(token);
+         fJoinFieldValues.emplace_back(ptr.get());
```

Comment:
With my comment on `REntry::Reset`, the pointer will not be constant anymore during processing. Also the user could call `REntry::EmplaceNewValue` themselves, in which case this might cause problems...